challenges/security/cloud-security.md

```diff
@@ -0,0 +1,241 @@
+ - Cloud provider specific security services such as:
+   - AWS: CloudTrail, GuardDuty, WAF
+
+This is asynchronous. You'll develop on your machine and you're welcome to
+use whatever tools you're comfortable with to develop the changes. We'll be
+available in the interview slack channel to answer any questions.
+
+We will review your PR, inculding running terraform plan. Make sure to write
+good commit messages, as well as a helpful PR description.
+
+# Part 5: Red Team & Blue Team discussion
+After your development PR is posted, you'll join a synchronous video call with
+future peers to finish the day. During this hour long discussion, we'll present
+you with a hypothetical service, and ask the following:
+
+1. How would you attack this service, specifically to exfiltrate the core data?
+2. How would you defend against the attacks you outlined earlier.
+
+At the end of this discussion, you'll have an opportunity to ask peers
+questions about working at Teleport.
+
+# Guidance
+
+## Interview process
+
+Although the interview is scheduled for a single day, the only in-person
+discussions are the video calls at the beginning and end of the day. Both the
+peer review and development portion are on your own machine without anyone
+hovering over your shoulder--as we work in our day-to-day at Teleport.
+
+Please make sure you have an environment set up for go development. The tools
+needed can be found in the [Tools](#Tools) section below.
+
+Should you have any questions during the day, please ask the interview team
+Slack channel.
+
+After you interview, a panel of engineers with review the submitted
+pull requests and vote using a "+1, -2" anonymous voting system: +1 is
+submitted whenever a team member accepts the submission, -2 otherwise.
+
+In case of a positive result, we will connect you to our HR and recruiting
+teams, who will work out the details and present an offer.
+
+In case of a negative score result, hiring manager will contact you and share a
+list of the key observations from the team that affected the result.
+
+We're actively improving this interview process, and we would be glad to hear
+your feedback about the experience.
+
+## Code and project ownership
+
+This is a contrived challenge and we have no intent of using the code you've
+submitted in production. However, we retain all rights to
+*our example code and RFD*. We ask that you do not publish our code to avoid
+giving future candidates an unfair advantage by allowing them develop a solution
+ahead of time.
+
+## Areas of focus
+
+These are the areas we will be evaluating in the submission:
+
+* Ensure the Terraform contains all resources needed.
+* Use consistent coding style. We follow `terraform fmt` and
+  [tfsec](https://github.com/aquasecurity/tfsec) for our Terraform.
+* Follow the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)
+  when configuring IAMs.
+* Show your breadth and depth of knowledge for either AWS or GCP.
```

Comment:
This is tricky.

We definitely want candidates with AWS experience.  GCP experience is nice, but not as useful as AWS given our architecture.  I don't think we'd be in a good position to review or hire Azure folks, or any of the smaller providers.

So that is one point in favor of sticking with AWS, and maybe allowing GCP as a 2nd option.

Beyond the skills we need, I'm also considering the candidate experience and how many different processes we have to create and review.  Consider:

It takes a couple interviews to get well calibrated on when candidates are good and when they're not as strong as others.  We're already developing and allowing app sec vs cloud sec processes, which won't be directly comparable.  If we allow cloud sec to fragment into AWS and GCP, we're decreasing the chance the data we'll have to directly compare candidates, and introducing a chance for more subjectivity to creep into the process. AWS and GCP aren't 1-to-1, and how do we evaluate a candidate prioritizing GCP specific features vs AWS specific features.  I wouldn't want to say "You can use GCP" but then have it cause candidates to start off on a worse foot than those who chose AWS

The other element is the investment of developing architecture and example codebases.  It will take additional work to put together example code for a GCP version in addition to an AWS version.  Given that we're already months behind on starting to hire for these roles, I'm looking to cut scope rather than increase it.

We can avoid the cost of additional example code dev by having the terraform challenge be greenfield (no starting example code), but my gut says it is top much ask a candidate to do greenfield development in 3 hours.  If we went with the async challenge, I'd feel better about greenfield development, and thus allowing GCP.

@reedloden  and I  discussed allowing candidates to choose a synchronous or asyncrhonous process. However I'm reluctant to do that in this PR, as it would compound the difficulty of comparing candidates who choose different options.  Given that we're hiring 4 security engineers, we'd quickly reach the point where we could have only one or two candidates in any given process.

In short: Today, I'm developing the process only for AWS.  We may iterate on it to add GCP in the near future.

Reply:
:+1: to just focusing on AWS for now.